CHANGELOG.md

```diff
@@ -97,6 +97,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [Refactor] XContent base classes from xcontent to core library ([#5902](https://github.com/opensearch-project/OpenSearch/pull/5902))
 
 ### Deprecated
+- Map, List, and Set in org.opensearch.common.collect ([#6609](https://github.com/opensearch-project/OpenSearch/pull/6609))
```

Comment:
This will be an invasive change across the project. Most plugins use these data structures[1].
I would take this opportunity to remove these implementations completely in 3.0 and have these deprecations in 2.x. Since we have the flexibility to make breaking changes in a new major. WDYT @reta @andrross @nknize 

[1] https://github.com/search?q=org%3Aopensearch-project%20org.opensearch.common.collect&type=code

Reply:
@saratvemulapalli it makes sense but the only concern I have is how far are we from 3.0.0 release? We have to give people (plugin developers primarily) enough time to move off the deprecated APIs, if 3.0.0 is year from now - this is fine o deprecate + remove, if just few months - we should postpone removal to 4.0.0 I think.

Reply:
These deprecations have already been merged as a part of `2.x`. I am trying to get main in sync with those changes.
We can have a campaign for removal, if necessary. But I still think we need to merge this in to get the branches in sync first.

Reply:
@reta from release schedule[1], looks like we are atleast 9-11 months away from 3.0. 
@kotwanikunal absolutely, lets get this merged.

If all 3 of us agree, could you create a campaign that these data structures will be removed?
Please Vote 👍🏻 👎🏻  

[1] https://opensearch.org/releases.html

Reply:
I will keep a track on the votes, but as per the current majority, I will create a campaign soon for this.

Reply:
Campaign: https://github.com/opensearch-project/OpenSearch/issues/6786